shared-libs/cht-datasource/src/person.ts

```diff
@@ -109,5 +92,50 @@ export namespace v1 {
    * @returns a function for retrieving a paged array of people
    * @throws Error if a data context is not provided
    */
-  export const getPage = getPeople(Local.Person.v1.getPage, Remote.Person.v1.getPage);
+  export const getPage = (
+    context: DataContext
+  ): ({ personType, limit, skip }: GetPageParams) => Promise<Person[]> => {
+    assertDataContext(context);
+    const fn = adapt(context, Local.Person.v1.getPage, Remote.Person.v1.getPage);
+
+    /**
+     * Returns an array of people for the provided page specifications.
+     * @param params the function params
+     * @param params.personType the type of people to return
+     * @param params.limit the maximum number of people to return. Default is 100.
+     * @param params.skip the number of people to skip. Default is 0.
+     * @returns an array of people for the provided page specifications.
+     * @throws Error if no type is provided or if the type is not for a person
+     * @throws Error if the provided `limit` value is `<=0`
+     * @throws Error if the provided `skip` value is `<0`
+     */
+    const curriedFn = async ({ personType, limit = 100, skip = 0}: GetPageParams): Promise<Person[]> => {
```

Comment:
Yeah, that was the initial implementation. However, `getDocumentStream` needs to call the function passed in function with the passed-in parameter too. The initial implementation included spreading the passed parameters without modifying the function in question but as @m5r flagged that would lead to having the consumer of the function `getDocumentStream` have to pass the arguments in an ordered way. [Ref](https://github.com/medic/cht-core/pull/9281#discussion_r1695425097). Since `getPage` is public whereas `getDocumentStream` is not, makes the consumers of `getDocumentStream` ourselves which might make us aware that we need to send in the params in order.

Reply:
So, ultimately I think we should still be able to get this to work via generics.  One think that should make our current code more simple is that [we only need to pass the person type](https://github.com/medic/cht-core/pull/9281#discussion_r1705349811) to `getDocumentStream` (and not `limit` or `skip`).  

I recognize that this may cause trouble in the future when we want to reuse this with a function that needs multiple parameters, but honestly I would rather take that risk than have to change our public `getPage` api just to support this now.  (Especially if we don't actually need the change.)

(Also I apologize for bringing this up late in the game here. :sweat: I did not fully realize the implications of Mokhtar's earlier suggestion until I saw it  fully coded up here...)